src/Parser.php

```diff
@@ -2559,39 +2559,47 @@ private function parseNamespaceUseDeclaration($parentNode) {
 
         $namespaceUseDeclaration->useKeyword = $this->eat(TokenKind::UseKeyword);
         $namespaceUseDeclaration->functionOrConst = $this->eatOptional(TokenKind::FunctionKeyword, TokenKind::ConstKeyword);
-        $namespaceUseDeclaration->namespaceName = $this->parseQualifiedName($namespaceUseDeclaration);
-        if (!$this->checkToken(TokenKind::OpenBraceToken)) {
+        if($this->lookahead(TokenKind::CommaToken)) {
```

Comment:
> So with these node revisions, we should end up with NamespaceUseDeclaration having useClauses, that always assumes theres a list (NamespaceUseClauseList), even with just one member? 

Yep! That said, it's not the best representation when it comes to memory usage. In the future, we may want to avoid defining properties that would be null, arrays with zero or one elements, etc. But that's an optimization that would apply to many Node types, not just this one.

> If so, whats the purpose of groupClauses inside the new NamespaceUseClause?

groupClauses refers to the delimited list that appears inside of the open braces and appends the namespaceName (`use A\B\{C\D,E\F}`), and useClauses refers a higher level (non-prepended) list of usings (`use A\B, C\D`)

Now, this is only one idea, and there are certainly other ways that you could represent this - the primary constraints being:
* higher level constructs are applied at the right scopes. For instance the function keyword in `use function A, B` should be "applied" to both child A and B (if you're at B, you should be able to inspect the parent node to see if it belongs in the function import table - you shouldn't have to inspect the first element in the list to know that.)
* all forms of usings defined in [the grammar](https://github.com/php/php-langspec/blob/master/spec/19-grammar.md#user-content-grammar-namespace-use-declaration) are represented, and in a way that makes it possible to add simple post-parse rules to produce any additional errors.
* all [invariants](https://github.com/Microsoft/tolerant-php-parser/blob/master/docs/Invariants.md) about the tree are satisfied.

Hope that helps, and feel free to prod further - there are a lot of nuances to this problem in particular, and I don't claim this is the perfect solution - so the more questions we ask, the better.